parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -445,6 +445,16 @@ EOF
         echo $(date),$(hostname), "End configuring GPU drivers"
     fi
 
+    export -f enable_managed_gpu_experience
+    ENABLE_MANAGED_GPU_EXPERIENCE=$(retrycmd_silent 10 1 10 bash -cx enable_managed_gpu_experience)
+    if [ "$?" -ne 0 ] && [ "${GPU_NODE}" = "true" ] && [ "${skip_nvidia_driver_install}" != "true" ]; then
+        echo "failed to determine if managed GPU experience should be enabled by nodepool tags"
+        exit $ERR_LOOKUP_ENABLE_MANAGED_GPU_EXPERIENCE_TAG
+    elif [ "${GPU_NODE}" = "true" ] && [ "${skip_nvidia_driver_install}" != "true" ] && [ "${ENABLE_MANAGED_GPU_EXPERIENCE}" = "true" ]; then
+        logs_to_events "AKS.CSE.installNvidiaDCGMPkgFromCache" "installNvidiaDCGMPkgFromCache" || exit $ERR_NVIDIA_DCGM_INSTALL
```

Comment:
what about  installNVidiaDevicePlugin ? should we combined both logic under the same condititions ? (if statement ?)

Reply:
Ideally both of these should be under the same tag. The whole installation of the deviceplugin and DCGM should be handled under one installation!

Reply:
Please see the latest commit, took care of this.